File: Visual Basic/ImportCSV.vb
'CSV File Without Headers
Private Function GetDataTableFromCSV(ByVal filename As String) As DataTable
    Dim CsvData As DataTable = New DataTable()

    Try
        Using csvReader As TextFieldParser = New TextFieldParser(filename)
            csvReader.SetDelimiters(New String() {";"})
            csvReader.HasFieldsEnclosedInQuotes = True
            
            'Add Columns
            CsvData.Columns.AddRange(New DataColumn() {
                New DataColumn("Column Name here", GetType(String))
            })

            'Read data from CSV
            While Not csvReader.EndOfData
                Dim fieldData As String() = csvReader.ReadFields()
                Dim dr As DataRow = CsvData.NewRow()

                For i As Integer = 0 To fieldData.Length - 1
                    If fieldData(i) = Nothing Then
                        'Null value found in CSV.
                        'Add error handeling
                    Else
                        dr(i) = fieldData(i)
                    End If
                Next
            End While
        End Using
    Catch ex As Exception
    
    End Try

    Return CsvData
End Function


'CSV File With Headers
Private Function GetDataTableFromCSV(ByVal filename As String) As DataTable
    Dim CsvData As DataTable = New DataTable()

    Try
        Using csvReader As TextFieldParser = New TextFieldParser(filename)
            csvReader.SetDelimiters(New String() {";"})
            csvReader.HasFieldsEnclosedInQuotes = True

            'Read Columns from CSV
            Dim colFieds As String() = csvReader.ReadFields()
            
            For Each column As String in colFieds
                Dim datacolumn As DataColumn = new DataColumn(column)
                CsvData.Columns.Add(datacolumn)
            Next

            'Read data from CSV
            While Not csvReader.EndOfData
                Dim fieldData As String() = csvReader.ReadFields()
                Dim dr As DataRow = CsvData.NewRow()

                For i As Integer = 0 To fieldData.Length - 1
                    If fieldData(i) = Nothing Then
                        'Null value found in CSV.
                        'Add error handeling
                    Else
                        dr(i) = fieldData(i)
                    End If
                Next
            End While
        End Using
    Catch ex As Exception
    
    End Try

    Return CsvData
End Function

File: CSharp/ImportCSV.cs
// CSV File Without Headers
namespace ImportCSV
{
    public partial class x 
    {
        private DataTable GetDataTableFromCSV(string filename)
        {
            DataTable CsvData = new DataTable();

            try {
                using (TextFieldParser csvReader = new TextFieldParser(filename)) {
                    csvReader.SetDelimiters(new string[] { ";" });
                    csvReader.HasFieldsEnclosedInQuotes = true;

                    // Add Columns
                    CsvData.Columns.AddRange(new DataColumn[] {
                        new DataColumn("Column Name here", typeof(string))
                    });

                    // Read data from CSV
                    while (!csvReader.EndOfData) {
                        string[] fieldData = csvReader.ReadFields();
                        DataRow dr = CsvData.NewRow();

                        for (int i = 0; i <= fieldData.Length - 1; i++) {
                            if (fieldData[i] == null) {
                                //Null value in csv found
                            }
                            else{
                                dr(i) = fieldData[i];
                            }
                        }
                    }
                }
            }
            catch (Exception ex) {
                throw;
            }

            return CsvData;
        }

        // CSV File With Headers
        private DataTable GetDataTableFromCSV(string filename)
        {
            DataTable CsvData = new DataTable();

            try {
                using (TextFieldParser csvReader = new TextFieldParser(filename)) {
                    csvReader.SetDelimiters(new string[] { ";" });
                    csvReader.HasFieldsEnclosedInQuotes = true;

                    // Read Columns from CSV
                    string[] colFieds = csvReader.ReadFields();

                    foreach (string column in colFieds) {
                        DataColumn datacolumn = new DataColumn(column);
                        CsvData.Columns.Add(datacolumn);
                    }

                    // Read data from CSV
                    while (!csvReader.EndOfData) {
                        string[] fieldData = csvReader.ReadFields();
                        DataRow dr = CsvData.NewRow();

                        for (int i = 0; i <= fieldData.Length - 1; i++) {
                            if (fieldData[i] == null) {
                                //Null value found in csv
                            }
                            else {
                                dr(i) = fieldData[i];
                            }
                        }
                    }
                }
            }
            catch (Exception ex) {
                throw;
            }

            return CsvData;
        }
    }
}
